engine-sdk/src/near_runtime.rs

```diff
@@ -365,6 +365,53 @@ impl crate::promise::PromiseHandler for Runtime {
     }
 }
 
+/// Some host functions are not usable in NEAR view calls.
+/// This struct puts in default values for those calls instead.
+pub struct ViewEnv;
+
+impl crate::env::Env for ViewEnv {
+    fn signer_account_id(&self) -> AccountId {
+        AccountId::new("system").unwrap()
+    }
+
+    fn current_account_id(&self) -> AccountId {
+        unsafe {
+            exports::current_account_id(Runtime::ENV_REGISTER_ID.0);
+        }
+        Runtime::read_account_id()
+    }
+
+    fn predecessor_account_id(&self) -> AccountId {
+        AccountId::new("system").unwrap()
+    }
+
+    fn block_height(&self) -> u64 {
+        unsafe { exports::block_index() }
+    }
+
+    fn block_timestamp(&self) -> crate::env::Timestamp {
+        let ns = unsafe { exports::block_timestamp() };
+        crate::env::Timestamp::new(ns)
+    }
+
+    fn attached_deposit(&self) -> u128 {
+        1
+    }
```

Comment:
Is there any particular reason to set `attached_deposit` as 1? Even though it's not used anywhere, I'd expect to see 0 here as otherwise, it looks a little bit misleading. Please let me know if I'm mistaken.

```suggestion
    fn attached_deposit(&self) -> u128 {
        0
    }
```

Reply:
I chose 1 because some functions ([for example `withdraw`](https://github.com/aurora-is-near/aurora-engine/blob/master/engine/src/lib.rs#L568)) require it to run and I wanted to make sure those functions could be called as well using this struct (as of now that isn't possible because there is no way for the environment to check if it is a view call or not, but maybe in the future there will be). There is no risk here because view calls do not commit to the chain, so there is no way for someone to abuse that 1 yoctoNEAR.